pkg/deploy/lattice/service_manager.go

```diff
@@ -2,318 +2,322 @@ package lattice
+	if err != nil {
+		return emtpyStatus, err
+	}
 
+	for _, snName := range svc.Spec.ServiceNetworkNames {
+		err = m.createAssociation(ctx, createSvcResp.Id, snName)
+		if err != nil {
+			return emtpyStatus, err
 		}
-
 	}
-	err = s.serviceNetworkAssociationMgr(ctx, service.Spec.ServiceNetworkNames, serviceID)
+	status := svcStatusFromCreateSvcResp(createSvcResp)
+	return status, nil
+}
 
+func (m *defaultServiceManager) createAssociation(ctx context.Context, svcId *string, snName string) error {
+	sn, err := m.datastore.GetServiceNetworkStatus(snName, m.cloud.Config().AccountId)
 	if err != nil {
-		return latticemodel.ServiceStatus{ServiceARN: "", ServiceID: ""}, err
+		return err
 	}
-
-	return latticemodel.ServiceStatus{ServiceARN: serviceArn, ServiceID: serviceID, ServiceDNS: serviceDNS}, nil
+	assocReq := &CreateSnSvcAssocReq{
+		ServiceIdentifier:        svcId,
+		ServiceNetworkIdentifier: aws.String(sn.ID),
+	}
+	assocResp, err := m.cloud.Lattice().CreateServiceNetworkServiceAssociationWithContext(ctx, assocReq)
+	if err != nil {
+		return err
+	}
+	err = handleCreateAssociationResp(assocResp)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-// TODO: Further refactor for all manager under lattice/
-func (s *defaultServiceManager) isServiceNetworkServiceAssociationStatusAssociated(associationStatus string) bool {
-	switch associationStatus {
-	case vpclattice.ServiceNetworkServiceAssociationStatusActive:
-		return true
-	case vpclattice.ServiceNetworkServiceAssociationStatusCreateInProgress:
-		return false
-	case vpclattice.ServiceNetworkServiceAssociationStatusDeleteFailed:
-		return false
-	case vpclattice.ServiceNetworkServiceAssociationStatusCreateFailed:
-		return false
-	case vpclattice.ServiceNetworkServiceAssociationStatusDeleteInProgress:
-		return false
-	default:
-		return false
+func (m *defaultServiceManager) newCreateSvcReq(svc *Service) *CreateSvcReq {
+	svcName := svc.LatticeName()
+	req := &vpclattice.CreateServiceInput{
+		Name: &svcName,
+	}
+
+	if svc.Spec.CustomerDomainName != "" {
+		req.CustomDomainName = &svc.Spec.CustomerDomainName
+	}
+	if svc.Spec.CustomerCertARN != "" {
+		req.SetCertificateArn(svc.Spec.CustomerCertARN)
 	}
+
+	return req
 }
 
-// find service by name return serviceNetwork,err if mesh exists, otherwise return nil,nil
-func (s *defaultServiceManager) findServiceByName(ctx context.Context, serviceName string) (*vpclattice.ServiceSummary, error) {
-	latticeSess := s.cloud.Lattice()
-	servicesListInput := vpclattice.ListServicesInput{}
-	resp, err := latticeSess.ListServicesAsList(ctx, &servicesListInput)
-	glog.V(6).Infof("findServiceByName, resp %v, err: %v\n", resp, err)
-
-	if err == nil {
-		for _, r := range resp {
-			if aws.StringValue(r.Name) == serviceName {
-				glog.V(6).Info("service ", serviceName, " already exists with arn ", *r.Arn, "\n")
-				return r, err
-			}
-		}
-	} else {
-		return nil, err
+func svcStatusFromCreateSvcResp(resp *CreateSvcResp) ServiceStatus {
```

Comment:
I see somewhere else similar code is used, I would suggest changing the signature to take the arn, id, and dns - I don't think for example the ```resp``` value here can be nil and identical logic lives in ```updateServiceAndAsociations```.